clang/lib/DPCT/APINamesThrust.inc

```diff
@@ -1416,3 +1416,47 @@ CONDITIONAL_FACTORY_ENTRY(
 CALL_FACTORY_ENTRY("thrust::make_zip_iterator",  CALL("oneapi::dpl::make_zip_iterator",  ARG(0)))
 // thrust::make_reverse_iterator
 CALL_FACTORY_ENTRY("thrust::make_reverse_iterator",  CALL("oneapi::dpl::make_reverse_iterator",  ARG(0)))
+
+// thrust::exclusive_scan
+CONDITIONAL_FACTORY_ENTRY(
+  CheckArgCount(3),
+  // thrust::exclusive_scan(data, data + 6, data)
+  CALL_FACTORY_ENTRY("thrust::exclusive_scan",
+    CALL("std::exclusive_scan",
+      CALL("oneapi::dpl::execution::make_device_policy", QUEUESTR),
+      ARG(0), ARG(1), ARG(2), ARG("0"))),
+  CONDITIONAL_FACTORY_ENTRY(
+    CheckArgCount(4),
+    CONDITIONAL_FACTORY_ENTRY(
+      CompareArgType(0, 1),
+      // thrust::exclusive_scan(thrust::host, data, data + 6, data);
+      CALL_FACTORY_ENTRY("thrust::exclusive_scan",
+        CALL("std::exclusive_scan",
+          makeMappedThrustPolicyEnum(0), ARG(1), ARG(2), ARG(3), ARG("0"))),
+      // thrust::exclusive_scan(data, data + 6, data, 4);
+      CALL_FACTORY_ENTRY("thrust::exclusive_scan",
```

Comment:
Need add additional condition to check the iterator is host/device (like line 1384)